tests/testthat/test_agg.R

```diff
@@ -0,0 +1,435 @@
+setkeyv(expected_dt, id_cols)
+
+description <- "aggregating an interval variable with varying interval id cols
+works"
+testthat::test_that(description, {
+  output_dt <- agg(
+    dt = input_dt,
+    id_cols = id_cols, value_cols = value_cols,
+    col_stem = "age", col_type = "interval",
+    mapping = age_mapping,
+    present_agg_severity = "skip"
+  )
+  testthat::expect_identical(output_dt, expected_dt)
+})
+
+description <- "throws error when aggregating an interval variable and there
+are missing intervals"
+testthat::test_that(description, {
+
+  new_input_dt <- input_dt[!(sex == "female" & (age_start <= 2 | age_end >= 95))]
+  setkeyv(new_input_dt, id_cols)
+
+  new_expected_dt <- expected_dt[!(sex == "female" & (age_start <= 2 | age_end >= 95))]
+  setkeyv(new_expected_dt, id_cols)
+
+  # check severity
+  expect_error(
+    agg(dt = new_input_dt,
+        id_cols = id_cols, value_cols = value_cols,
+        col_stem = "age", col_type = "interval",
+        mapping = age_mapping,
+        missing_dt_severity = "stop",
+        present_agg_severity = "skip"
+    ),
+    regexp = "intervals in `dt` are missing making it impossible to collapse"
+  )
+
+  output_dt <- agg(
+    dt = new_input_dt,
+    id_cols = id_cols, value_cols = value_cols,
+    col_stem = "age", col_type = "interval",
+    mapping = age_mapping,
+    missing_dt_severity = "none",
+    present_agg_severity = "skip"
+  )
+  expect_identical(output_dt, new_expected_dt)
+})
+
+description <- "error is thrown when interval aggregate is already included
+in input or works when `present_agg_severity = 'none'`"
+testthat::test_that(description, {
+
+  new_input_dt <- unique(rbind(input_dt, expected_dt))
+  setkeyv(new_input_dt, id_cols)
+
+  setkeyv(expected_dt, id_cols)
+
+  expect_error(
+    agg(dt = new_input_dt,
+        id_cols = id_cols, value_cols = value_cols,
+        col_stem = "age", col_type = "interval",
+        mapping = age_mapping
+    ),
+    regexp = "overlapping intervals were identified in `dt`"
+  )
+})
+
+
+# Aggregate categorical variable with multiple levels in mapping ----------
+
+# Inputs:
+# - present day provinces (only the most detailed nodes) in Iran as defined in `iran_mapping`
+#   see https://ihmeuw-demographics.github.io/hierarchyUtils/articles/hierarchyUtils.html#aggregate-locations-1
+#   for visualization of mapping
+# Output:
+# - all historical provinces and the national level
+
+id_cols <- c("location", "year")
+value_cols <- c("value")
+
+# set up test input data.table with only the present day provinces
+input_dt <- CJ(
+  location = iran_mapping[!grepl("[0-9]+", child), child],
+  year = 2011,
+  value = 1
+)
+setkeyv(input_dt, id_cols)
+
+# set up expected output table with all unique locations
+# the expected value is the number of leaf nodes under each aggregate
+expected_dt <- CJ(location = unique(iran_mapping$parent),
+                  year = 2011)
+expected_dt[location %in% c("Tehran 2006", "Zanjan 1976-1996",
```

Comment:
Is this the correct way to format long lines like these?

Reply:
I think it would be 
```
expected_dt[
         location %in% c(
                 "Tehran 2006", ............ etc
                  ),
          value := 2
]
```
?

Reply:
Good point, thanks!